flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/message/PulsarMessageBuilder.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.pulsar.sink.writer.message;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.connector.pulsar.sink.writer.router.KeyHashTopicRouter;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.TypedMessageBuilder;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** {@link TypedMessageBuilder} wrapper for providing the required method for end users. */
+@PublicEvolving
+public class PulsarMessageBuilder {
+
+    private byte[] orderingKey;
+    private String key;
+    private long eventTime;
+    Schema<?> schema;
+    private Object value;
+    private Map<String, String> properties = new HashMap<>();
+    private Long sequenceId;
+    private List<String> replicationClusters;
+    private boolean disableReplication = false;
+
+    /** Method wrapper of {@link TypedMessageBuilder#orderingKey(byte[])}. */
+    public PulsarMessageBuilder orderingKey(byte[] orderingKey) {
+        this.orderingKey = checkNotNull(orderingKey);
+        return this;
+    }
+
+    /**
+     * Property {@link TypedMessageBuilder#key(String)}. This property would also be used in {@link
+     * KeyHashTopicRouter}.
+     */
+    public PulsarMessageBuilder key(String key) {
+        this.key = checkNotNull(key);
+        return null;
+    }
+
+    /** Method wrapper of {@link TypedMessageBuilder#eventTime(long)}. */
+    public PulsarMessageBuilder eventTime(long eventTime) {
+        this.eventTime = eventTime;
+        return this;
+    }
+
+    /**
+     * Method wrapper of {@link TypedMessageBuilder#value(Object)}. We would use default {@link
+     * Schema#BYTES} if you use this method.
+     */
+    public PulsarMessageBuilder value(byte[] value) {
+        this.schema = Schema.BYTES;
+        this.value = value;
+        return this;
+    }
```

Comment:
Nit: Maybe it is easier to only offer the other builder method `value(Schema<IN> schema, IN value)` and move the `Schema.BYTES` delegation to the `PulsarSerializationSchemaWrapper` and `PulsarSchemaWrapper`.
Otherwise you should probably add some checks that only one of the value methods is called.